### PR TITLE
Rename `edgedb self` to `edgedb cli`

### LINKED ISSUE #416: Rename `edgedb self` to `edgedb cli`
(see https://github.com/edgedb/rfcs/blob/master/text/1006-simplified-cli.rst)

### PR DESCRIPTION
**Pull request by fantix**
Also reorganized `edgedb cli` related source files into `crate::cli` package.

The hidden command `edgedb _self_install` is kept unchanged to be compatible with the installation script at https://sh.edgedb.com.

Fixes #416

### CODE REVIEW

File: src/cli/cli_install.rs [hunk 1]
```diff
@@ -94,7 +94,7 @@ pub struct Settings {
 fn print_long_description(settings: &Settings) {
     println!();
     print_markdown!("\
-        # Welcome to EdgeDB!\n\
```
Comment (tailhook) [inline src/cli/cli_install.rs:97]: This is interesting. Does it have a better style? We should probably change the style instead?

File: src/cli/mod.rs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+pub mod cli_install;
```
Comment (tailhook) [inline src/cli/mod.rs:1]: I would remove `cli_` from module names here. Looks like a duplication.

File: src/options.rs [hunk 3]
```diff
@@ -362,6 +343,23 @@ fn get_deprecated_matches(mismatch_cmd: &str) -> Option<clap::ArgMatches> {
             }
             _ => return None,
         }
+        Some(first_cmd) if first_cmd == "self" => match args.next() {
```
Comment (tailhook) [inline src/options.rs:346]: Shouldn't we skip `self` fallback, because there is no release which had these working?

**Review by tailhook — APPROVED** [review]

File: src/options.rs (hunk 3 above)
Comment (fantix) [reply src/options.rs:346]: Ah yes, fixed!

File: src/cli/cli_install.rs (hunk 1 above)
Comment (fantix) [reply src/cli/cli_install.rs]: Found a way to fix the alignment.

**Pull request merged**